From 8ead9473d7ee5c8e7348a30127db800ed82dde8a Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Wed, 17 Feb 2021 16:03:44 +0200
Subject: [PATCH] [SW-20154] gaudi2: add the actual key hashes

Add the key hash computed on the Public key Properties
of the keys generated for Gaudi2.

Change-Id: I42571a54b8f1fbd0e152d3c8184c967b3251e915
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/gaudi2_arc_gnu/secureboot.c b/gaudi2_arc_gnu/secureboot.c
index 8a3a60e..ea11bc4 100644
--- a/gaudi2_arc_gnu/secureboot.c
+++ b/gaudi2_arc_gnu/secureboot.c
@@ -24,45 +24,49 @@
 const uint8_t rom_key_hashes[NUM_KEYS_IN_ROM][SHA384_SUM_LEN] = {
 	/* Habana Public Key */
 	{
-		0xa6, 0xec, 0xfe, 0x38, 0x44, 0x0b, 0x8b, 0xf2,
-		0xef, 0x96, 0xef, 0x2f, 0x8d, 0xcb, 0xf1, 0x5a,
-		0x76, 0x4a, 0xdf, 0x3a, 0xda, 0xce, 0x33, 0x03,
-		0x67, 0x30, 0x32, 0xfd, 0xda, 0x74, 0xd0, 0x6d,
-		0x64, 0x1b, 0x00, 0x92, 0x4c, 0x07, 0xad, 0xd5,
-		0xfc, 0xde, 0x0c, 0xba, 0xc1, 0xbd, 0x8b, 0x30
+		0x08, 0x51, 0xe7, 0x07, 0xc1, 0x0a, 0xc8, 0x45,
+		0x51, 0x61, 0xb4, 0xdf, 0xb6, 0xd6, 0x89, 0xbc,
+		0x57, 0xbb, 0xf9, 0xe2, 0x50, 0x4f, 0x44, 0x1f,
+		0x12, 0x4a, 0x73, 0x3d, 0xbb, 0xda, 0x63, 0xf2,
+		0x76, 0x53, 0x6c, 0x12, 0xa1, 0x13, 0x37, 0x30,
+		0x03, 0xee, 0x79, 0xce, 0x20, 0xa3, 0x32, 0x33
 	},
-	/* Additional Keys - TODO */
+	/* Additional Keys*/
+	/* Production Key 1 */
 	{
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0
+		0x47, 0x9b, 0x51, 0x5d, 0xf9, 0x8b, 0xfd, 0x3d,
+		0x38, 0x3f, 0x19, 0x51, 0x40, 0x62, 0xcb, 0x15,
+		0x2d, 0xae, 0x11, 0xe8, 0x1a, 0x91, 0x3e, 0x02,
+		0xa0, 0x80, 0x08, 0x9a, 0x62, 0x93, 0xb7, 0x65,
+		0x73, 0xe1, 0x6f, 0x86, 0x32, 0x16, 0xea, 0x37,
+		0x48, 0x19, 0xb8, 0xe0, 0xc3, 0x3b, 0x5a, 0xcf
 	},
+	/* Production Key 2 */
 	{
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0
+		0x48, 0x3c, 0x50, 0xfc, 0xf1, 0x3a, 0x56, 0x7a,
+		0x7b, 0x7c, 0xcc, 0x77, 0x66, 0xb3, 0x85, 0x1b,
+		0x6f, 0x43, 0x60, 0xce, 0x16, 0x9a, 0xc4, 0x14,
+		0xd5, 0xcb, 0x11, 0x09, 0x66, 0xc0, 0xfe, 0x42,
+		0xfe, 0x78, 0x79, 0x57, 0xdb, 0x06, 0x5c, 0x37,
+		0x2f, 0xe2, 0x81, 0xb0, 0x00, 0x11, 0x9f, 0x8e,
 	},
+	/* Production Key 3 */
 	{
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0
+		0x5c, 0xc8, 0xd6, 0xf3, 0x42, 0xdc, 0xa3, 0xdf,
+		0xb0, 0xa3, 0xa8, 0x73, 0x68, 0x54, 0x78, 0x5e,
+		0x8c, 0xaf, 0x09, 0x0b, 0x0d, 0x41, 0xa8, 0xa1,
+		0xd9, 0xf8, 0x31, 0xdc, 0x95, 0x0f, 0x84, 0x02,
+		0xb3, 0x2a, 0x61, 0xe4, 0x10, 0xd8, 0x39, 0xdf,
+		0x80, 0x27, 0xc1, 0x2e, 0xd5, 0xd2, 0xb2, 0xcd,
 	},
+	/* Production Key 4 */
 	{
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-		0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0
+		0xdb, 0x8c, 0x8c, 0xe7, 0x08, 0x19, 0x6b, 0x04,
+		0x76, 0x43, 0xa7, 0xe4, 0x34, 0x2d, 0x2e, 0x8f,
+		0xf5, 0xdd, 0x20, 0xc9, 0xc7, 0x48, 0x0c, 0x4d,
+		0x80, 0x35, 0x18, 0x10, 0x8b, 0xb2, 0x04, 0xd2,
+		0x79, 0xc1, 0x6b, 0x33, 0x07, 0x99, 0x21, 0x3a,
+		0xbf, 0x77, 0xfa, 0x6b, 0x53, 0xf4, 0x27, 0x25,
 	}
 };
 
